src/mgmt/shared/ranks.pb-c.c

```diff
@@ -0,0 +1,157 @@
+/* Generated by the protocol buffer compiler.  DO NOT EDIT! */
+/* Generated from: ranks.proto */
```

Comment:
So in this change there aren't any updates to the proto-definitions in ranks.proto, it's just included in a new location hence this compiled file being added. Is that right?

Reply:
Right, @tanabarr. This PR needs to use some protobuf stuff that the C side isn't currently using.

By the way, I had this thought, when fighting with these files, that how about reorganizing the generated C files into a central directory/module, and all C users link with this directory/module? Also, in the long run maybe we should remove generated protobuf files from Git, to avoid conflicts over them, protoc version inconsistency, and security concerns (it's hard to review generated code). If any of these sounds good to you, I might propose formally to you and Kris after 2.8 feature-freezes.

Reply:
I've got no objections to either suggestion.

Reply:
Although removing the generated files from git may have other related issues if people have to build them through scons.

Reply:
@kjacque, thanks for the comment. I'll create a ticket instead of a wiki page then, since the high-level idea is not as controversial as I expected.